exporters/stats/signalfx/src/main/java/io/opencensus/exporter/stats/signalfx/SignalFxSessionAdaptor.java

```diff
@@ -17,172 +17,145 @@
+      new Function<Long, Datum>() {
+        @Override
+        public Datum apply(Long arg) {
+          Datum.Builder builder = Datum.newBuilder();
+          builder.setIntValue(arg);
+          return builder.build();
+        }
+      };
+  private static final Function<Distribution, Datum> datumDistributionFunction =
+      new Function<Distribution, Datum>() {
+        @Override
+        public Datum apply(Distribution arg) {
+          // Signal doesn't handle Distribution value.
+          // TODO(mayurkale): decide what to do with Distribution value.
+          throw new IllegalArgumentException("Distribution type are not supported");
+        }
+      };
+  private static final Function<Summary, Datum> datumSummaryFunction =
+      new Function<Summary, Datum>() {
+        @Override
+        public Datum apply(Summary arg) {
+          // Signal doesn't handle Summary value.
+          // TODO(mayurkale): decide what to do with Summary value.
+          throw new IllegalArgumentException("Summary type are not supported");
+        }
+      };
+
   private SignalFxSessionAdaptor() {}
 
   /**
-   * Converts the given view data into datapoints that can be sent to SignalFx.
-   *
-   * <p>The view name is used as the metric name, and the aggregation type and aggregation window
-   * type determine the metric type.
+   * Converts the given Metric into datapoints that can be sent to SignalFx.
    *
-   * @param data The {@link ViewData} containing the aggregation data of each combination of tag
-   *     values.
-   * @return A list of datapoints for the corresponding metric timeseries of this view's metric.
+   * @param metric The {@link Metric} containing the timeseries of each combination of label values.
+   * @return A list of datapoints for the corresponding metric timeseries of this metric.
    */
-  static List<DataPoint> adapt(ViewData data) {
-    View view = data.getView();
-    List<TagKey> keys = view.getColumns();
-
-    MetricType metricType = getMetricTypeForAggregation(view.getAggregation(), view.getWindow());
+  static List<DataPoint> adapt(Metric metric) {
+    MetricDescriptor metricDescriptor = metric.getMetricDescriptor();
+    MetricType metricType = getType(metricDescriptor.getType());
     if (metricType == null) {
       return Collections.emptyList();
     }
 
-    List<DataPoint> datapoints = new ArrayList<>(data.getAggregationMap().size());
-    for (Map.Entry<List</*@Nullable*/ TagValue>, AggregationData> entry :
-        data.getAggregationMap().entrySet()) {
-      datapoints.add(
-          DataPoint.newBuilder()
-              .setMetric(view.getName().asString())
-              .setMetricType(metricType)
-              .addAllDimensions(createDimensions(keys, entry.getKey()))
-              .setValue(createDatum(entry.getValue()))
-              .build());
+    DataPoint.Builder shared = DataPoint.newBuilder();
+    shared.setMetric(metricDescriptor.getName());
+    shared.setMetricType(metricType);
```

Comment:
If `metricType` is `null` (`Summary` or `Distribution`), you can just return an empty list here.